internal/util/util.go

```diff
@@ -447,8 +449,14 @@ func GetModTime(path string) (time.Time, error) {
 	return info.ModTime(), nil
 }
 
-func AppendBackupSuffix(path string) string {
-	return path + ".micro-backup"
+func CreateTempBackupName(path string) string {
+	path += ".micro-backup"
+	name := filepath.Base(path)
+	if len(name) > fileNameLengthLimit {
+		dir := filepath.Dir(path)
+		path = filepath.Join(dir, HashStringMd5(path))
```

Comment:
Just the hash, without any extension?

Can't we just adjust `DetermineEscapePath()` to use `fileNameLengthLimit - len(".micro-backup")` instead of `fileNameLengthLimit` instead?

Reply:
You're right. No need to process this twice.
Only benefit was, to have a bit longer filename...now we hash earlier, even the escaped path doesn't exceed the max filename limit.